sailfishusermanager.cpp

```diff
@@ -382,6 +391,44 @@ void SailfishUserManager::setUserLimits(uint uid)
     }
 }
 
+bool SailfishUserManager::setUserUUID(uint uid)
+{
+    QFile file(USER_UUID_FILE);
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+        return false;
+
+    QTextStream in(&file);
+    QString content = in.readAll();
+    file.close();
+
+    QString new_content;
+    auto lines = content.split('\n');
+    for (auto line : lines) {
+        QStringList list = line.split(QLatin1Char(' '));
+        if (list.size() < 2)
+            continue;
+        uint id  = list[0].toUInt();
+        // skip add same uid in case of user recreation
+        if ( id != uid ) {
+            new_content += line;
+            new_content += '\n';
+        }
+    }
+    new_content += QString::number(uid);
+    new_content += ' ';
+    new_content += QUuid::createUuid().toString();
+    new_content += '\n';
+
+    QFile new_file("/etc/user_uuid");
+    if (!new_file.open(QIODevice::WriteOnly | QIODevice::Text))
+        return false;
+
+    QTextStream out(&new_file);
+    out << new_content;
+
```

Comment:
Missing new_file.close()

Reply:
I think this happens in dtor of qfile object, but if you want I can add this.

Reply:
This moves to libuser lib.